openedx_tagging/core/tagging/models.py

```diff
@@ -409,6 +413,32 @@ def tag_object(
 
         return updated_tags
 
+    def autocomplete_tags(self, search: str, object_id: str = None) -> models.QuerySet:
+        """
+        Returns tag values that contains the given search string.
+        The result is returned in alphabetical order.
+
+        The output is a QuerySet of dictionaries in with `_value` and `tag`.
+
+        Subclasses can override this method to perform their own autocomplete process.
+        """
+        # Fetch tags that the object already has to exclude them from the result
+        excluded_tags = self.objecttag_set.filter(object_id=object_id).values_list(
+            "_value", flat=True
+        )
+        return (
+            # Fetch object tags from this taxonomy whose value contains the search
+            self.objecttag_set.filter(_value__icontains=search)
+            # omit any tags whose values match the tags on the given object
+            .exclude(_value__in=excluded_tags)
+            # alphabetical ordering
+            .order_by("_value")
+            # obtain tag values
+            .values("_value", "tag")
```

Comment:
We can alias the `_value` field to `value` to make it a little nicer for users:

```suggestion
            # Alias the `_value` field to `value` to make it nicer for users
            .annotate(value=models.F('_value'))
            # obtain tag values and IDs
            .values("value", "tag")
```

Just need to update your API tests to use `tag.get("value")` instead of `tag.get("_value")`.

Reply:
Oh! Nice! :+1:

Reply:
Updated here: https://github.com/openedx/openedx-learning/pull/59/commits/593c047b8d7bf5c3a314b8a25d19f55beb5238c8